chunkset_tpl.h

```diff
@@ -42,46 +42,8 @@ Z_INTERNAL uint8_t* CHUNKCOPY(uint8_t *out, uint8_t const *from, unsigned len) {
 Z_INTERNAL uint8_t* CHUNKCOPY_SAFE(uint8_t *out, uint8_t const *from, unsigned len, uint8_t *safe) {
     unsigned safelen = (unsigned)((safe - out) + 1);
     len = MIN(len, safelen);
-#if CHUNK_SIZE >= 32
-    while (len >= 32) {
-        memcpy(out, from, 32);
-        out += 32;
-        from += 32;
-        len -= 32;
-    }
-#endif
-#if CHUNK_SIZE >= 16
-    while (len >= 16) {
-        memcpy(out, from, 16);
-        out += 16;
-        from += 16;
-        len -= 16;
-    }
-#endif
-#if CHUNK_SIZE >= 8
-    while (len >= 8) {
-        zmemcpy_8(out, from);
-        out += 8;
-        from += 8;
-        len -= 8;
-    }
-#endif
-    if (len >= 4) {
-        zmemcpy_4(out, from);
-        out += 4;
-        from += 4;
-        len -= 4;
-    }
-    if (len >= 2) {
-        zmemcpy_2(out, from);
-        out += 2;
-        from += 2;
-        len -= 2;
-    }
-    if (len == 1) {
-        *out++ = *from++;
-    }
-    return out;
+    memcpy(out, from, len);
```

Comment:
Can we just remove `chunkcopy_safe` altogether from `functable` and just replace with this code? What do you think?

Reply:
Sure, one less layer of indirection.  I have another thing I might also want to try since I saw a fair bit of the code was moving 16 and 8 bits at time into the vector register.  I can construct a permutation table and it'd be quite a bit faster for that.

Reply:
Errr well, we can do this but the problem is how we calculate len from safe.  This function is called from the table with 4 arguments, where the fourth is the last possible element position.  We need some sort of a wrapper for this.

Reply:
We could change every call of this to just compute what the actual len is in side the function it's being called from, but we'll need to do that everywhere.

Reply:
So, I've kept it as a function but one that is highly likely to be inlined away (it pretty much has to be, given that's declared as static in the two places it's used).  This just prevents us from duplicating the pointer arithmetic over and over where it's called.